budget-binder-multiplatform-app/src/commonMain/kotlin/de/hsfl/budgetBinder/domain/usecase/AuthUseCase.kt

```diff
@@ -18,17 +19,17 @@ class RegisterUseCase(private val repository: AuthRepository) {
                     emit(DataResponse.Success(it))
                 } ?: response.error?.let { error ->
                     when (error.code) {
-                        HttpStatusCode.Unauthorized.value -> emit(DataResponse.Unauthorized())
-                        else -> emit(DataResponse.Error(error.message))
+                        HttpStatusCode.Unauthorized.value -> emit(DataResponse.Unauthorized(error))
+                        else -> emit(DataResponse.Error(error))
                     }
                 }
             }
         } catch (e: IOException) {
             e.printStackTrace()
-            emit(DataResponse.Error("Couldn't reach the server"))
+            emit(DataResponse.Error(ErrorModel("Couldn't reach the server")))
         } catch (e: Throwable) {
             e.printStackTrace()
-            emit(DataResponse.Error(e.message.toString()))
+            emit(DataResponse.Error(ErrorModel("Somthing went wrong")))
```

Comment:
In allen Use Case sit ein Typo Somthing -> Something

Reply:
Oh no, da is wohl _somthing_ wrong gegangen! 😛 Sollte ja zum Glück leicht zu fixen sein